packages/marble-api/scripts/openapi.yaml

```diff
@@ -1657,6 +1657,78 @@ paths:
           $ref: '#/components/responses/403'
         '404':
           $ref: '#/components/responses/404'
+  /scenario-publications/preparation:
+    get:
+      tags:
+        - Scenario Publications
+      summary: Get the preparation status of a scenario publication
+      operationId: getScenarioPublicationPreparationStatus
+      security:
+        - bearerAuth: []
+      requestBody:
+        description: 'Describe the scenario publication action'
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - scenario_iteration_id
+              properties:
+                scenario_iteration_id:
+                  type: string
+                  format: uuid
+        required: true
+      responses:
+        '200':
+          description: The preparation status of the scenario publication
+          content:
+            application/json:
+              schema:
+                type: object
+                required:
+                  - preparation_status
+                properties:
+                  preparation_status:
+                    type: string
+                    enum:
+                      - 'PENDING'
+                      - 'IN_PROGRESS'
+                      - 'DONE'
```

Comment:
This is meant to change depending of what we really need but you have the idea: expose information based on current preparation status.
The granularity depends on :
- what we can really get from backend (ex: index creation status)
- what we really want to display to the user

Reply:
all right,
I'd suggest for now the following values:
```
- 'PREPARATION_REQUIRED'
- 'PREPARATION_IN_PROGRESS'
- 'READY_TO_ACTIVATE'
```
While keeping in mind the fact that we won't be able to tell for which iteration the (a) preparation is in progress

Reply:
I would suggest to distinguish things then :
```ts
type PreparationStatus = 'PREPARATION_REQUIRED' | 'PREPARATION_IN_PROGRESS' | 'READY_TO_ACTIVATE';

type PreparationServiceStatus = 'AVAILABLE' | 'OCCUPIED' // a temporary "global" state we need to use until a queue system or equivalent is introduced
```
NB: I know we can't return `'PREPARATION_IN_PROGRESS'` for now but it's a future possibility. It's just to illustrate we may need more than just a "boolean" state for that.

Reply:
agreed